---
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# PhantasusLite

PhantasusLite -- a tool designed to integrate the work with RNA-Seq count matrices into a single and fast R-pipeline. This R-package supports loading the data from GEO by GSE. It provides url access to the remote repository with the archs4, archs4_zoo and dee2 HDF5-files for getting the count matrix. Finally phantasusLite allows to get an ExpressionSet with the expression matrix for future differential expression analysis.

## Installation

To install the package you should:

1.  Install R-package rhdf5client from the github
2.  Install R-package phantasusLite from the github

```{r message=FALSE, eval=FALSE}
library(devtools)
install_github("assaron/rhdf5client")
install_github("ctlab/phantasusLite")
```

## Dependencies

To run the code you need:

-   `GEOquery`
-   `rhdf5client`
-   `phantasuslite`

```{r message=FALSE, warning=FALSE}
library(GEOquery)
library(rhdf5client)
library(phantasusLite)
```


## Quick start

To run the package enter the code sample below.

Let's load the ExpressionSet from GEO

```{r message=FALSE}
ess <- getGEO("GSE53053")
es <- ess[[1]]
```


ExpressionSet from the GEO doesn't contain the expression matrix -- `exprs(es)` is empty.

```{r}
head(exprs(es))
```

Function loadCountsFromHSDS returns an ExpressionSet with the expression matrix -- the second exprs(es) contains an expression matrix.

The remote repository URL is '<https://ctlab.itmo.ru/hsds/?domain=/counts>'.

```{r}
url <- 'https://ctlab.itmo.ru/hsds/?domain=/counts'
es <- loadCountsFromHSDS(es, url)
head(exprs(es))
```


The available gene annotations are also filled in:
```{r}
head(fData(es))
```
